packages/i18n/locale/locale.js

```diff
@@ -21,45 +20,70 @@ import {setResBundleLocale} from '../src/resBundle';
  * @memberof i18n/locale
  * @param {ilib.Locale|string|undefined} spec locale specifier or locale object of the locale to
  *	test, or undefined to test the current locale
- * @returns {Boolean} `true` if non-Latin locale
+ * @param {Object} [options] An object configuring the request. Must include an `onLoad` member to
+ *                           receive the result.
  */
-function isNonLatinLocale (spec) {
-	const li = new LocaleInfo(spec),
-		locale = li.getLocale();
+function isNonLatinLocale (spec, options = {}) {
+	const {onLoad} = options;
 
-	// We use the non-latin fonts for these languages (even though their scripts are technically
-	// considered latin)
-	const nonLatinLanguageOverrides =  ['vi', 'en-JP'];
-	// We use the latin fonts (with non-Latin fallback) for these languages (even though their
-	// scripts are non-latin)
-	const latinLanguageOverrides = ['ko', 'ha'];
+	if (!onLoad) return;
 
-	return (
-		(
-			// the language actually is non-latin
-			li.getScript() !== 'Latn' ||
-			// the language is treated as non-latin
-			nonLatinLanguageOverrides.indexOf(locale.getLanguage()) !== -1 ||
-			// the combination of language and region is treated as non-latin
-			nonLatinLanguageOverrides.indexOf(locale.toString()) !== -1
-		) && (
-			// the non-latin language should be treated as latin
-			latinLanguageOverrides.indexOf(locale.getLanguage()) < 0
-		)
-	);
+	// eslint-disable-next-line no-new
+	new LocaleInfo(spec, {
+		...options,
+		onLoad: (li) => {
+			const locale = li.getLocale();
+
+			// We use the non-latin fonts for these languages (even though their scripts are technically
+			// considered latin)
+			const nonLatinLanguageOverrides =  ['vi', 'en-JP'];
```

Comment:
It occurs to me we should document this (if we didn't?) and, possibly, move this override to `moonstone` to separate our non-standard stuff.

Reply:
Yeah. This should live in moonstone. I'll do a bit of refactoring to expose those as well.